packages/dma-library/src/operations/ajna/close-to-collateral.ts

```diff
@@ -0,0 +1,113 @@
+import { ajnaCloseToCollateralOperationDefinition } from '@deploy-configurations/operation-definitions'
+import { FEE_BASE, MAX_UINT, ZERO } from '@dma-common/constants'
+import { actions } from '@dma-library/actions'
+import { BALANCER_FEE } from '@dma-library/config/flashloan-fees'
+import {
+  IOperation,
+  WithAjnaBucketPrice,
+  WithAjnaStrategyAddresses,
+  WithCollateral,
+  WithDebt,
+  WithFlashloan,
+  WithProxy,
+  WithSwap,
+} from '@dma-library/types'
+import { FlashloanProvider } from '@dma-library/types/common'
+import BigNumber from 'bignumber.js'
+
+type AjnaCloseArgs = WithCollateral &
+  WithDebt &
+  WithSwap &
+  WithFlashloan &
+  WithProxy &
+  WithAjnaStrategyAddresses &
+  WithAjnaBucketPrice
+
+export type AjnaCloseToCollateralOperation = ({
+  collateral,
+  debt,
+  swap,
+  flashloan,
+  proxy,
+  addresses,
+  price,
+}: AjnaCloseArgs) => Promise<IOperation>
+
+export const closeToCollateral: AjnaCloseToCollateralOperation = async ({
+  collateral,
+  debt,
+  swap,
+  flashloan,
+  proxy,
+  addresses,
+  price,
+}) => {
+  const setDebtTokenApprovalOnPool = actions.common.setApproval({
+    asset: debt.address,
+    delegate: addresses.pool,
+    amount: flashloan.amount,
+    sumAmounts: false,
+  })
+
+  const paybackWithdraw = actions.ajna.ajnaPaybackWithdraw({
+    quoteToken: debt.address,
+    collateralToken: collateral.address,
+    withdrawAmount: ZERO,
+    paybackAmount: ZERO,
+    withdrawAll: true,
+    paybackAll: true,
+    price,
+  })
+
+  const swapCollateralTokensForDebtTokens = actions.common.swap({
+    fromAsset: collateral.address,
+    toAsset: debt.address,
+    amount: swap.amount,
+    receiveAtLeast: swap.receiveAtLeast,
+    fee: swap.fee,
+    withData: swap.data,
+    collectFeeInFromToken: swap.collectFeeFrom === 'sourceToken',
+  })
+
+  const unwrapEth = actions.common.unwrapEth({
+    amount: new BigNumber(MAX_UINT),
+  })
+
+  unwrapEth.skipped = !debt.isEth && !collateral.isEth
+
+  const sendQuoteTokenToOpExecutor = actions.common.sendToken({
+    asset: debt.address,
+    to: addresses.operationExecutor,
+    amount: flashloan.amount.plus(BALANCER_FEE.div(FEE_BASE).times(flashloan.amount)),
+  })
```

Comment:
in case when we were flashfloaning WETH, shouldn't we send it back to OP before unwrapping it?

Reply:
Yes absolutely. Great point.